src/createClassProxy.js

```diff
@@ -76,19 +76,10 @@ function proxyClass(InitialComponent) {
   }
 
   let displayName = getDisplayName(InitialComponent);
-  try {
-    // Create a proxy constructor with matching name
-    ProxyComponent = new Function('factory', 'instantiate',
-      `return function ${displayName}() {
-         return instantiate(factory, this, arguments);
-      }`
-    )(() => CurrentComponent, instantiate);
-  } catch (err) {
-    // Some environments may forbid dynamic evaluation
-    ProxyComponent = function () {
-      return instantiate(() => CurrentComponent, this, arguments);
-    };
-  }
+  ProxyComponent = function DynamicProxyComponent() {
```

Comment:
It's useful for HOCs to wrap the displayName of the base component, for example `HigherOrderComponent(BaseComponent)`. This is not a valid function name however, which causes unnecessary errors to be thrown.